src/io/aleph/dirigiste/Pools.java

```diff
@@ -34,7 +34,7 @@ public boolean shouldIncrement(Object key, int objectsForKey, int totalObjects)
             }
 
             public Map adjustment(Map stats) {
-                Map adj = new HashMap();
+                final Map adj = new HashMap(Stats.hashMapCapacity(stats.size()));
```

Comment:
I'd like to separate out these sorts of optimizations from the fixes elsewhere in this PR.  I understand the rationale behind trying to keep these maps as small as possible, but I'm not convinced it makes a meaningful difference, even in the crawling use case you have.

Reply:
Right, I'll remove them from this PR and submit a new one with only those changes.

Reply:
I've created a separate PR #20 with only the HashMap changes.